server/events/vcs/github_client.go

```diff
@@ -241,6 +259,53 @@ func (g *GithubClient) HidePrevCommandComments(repo models.Repo, pullNum int, co
 	return nil
 }
 
+func (g *GithubClient) getPRReviews(repo models.Repo, pull models.PullRequest) (GithubPRReviewSummary, error) {
+	var query struct {
+		Repository struct {
+			PullRequest struct {
+				ReviewDecision githubv4.String
+				Reviews        struct {
+					Nodes []GithubReview
+					// contains pagination information
+					PageInfo struct {
+						EndCursor   githubv4.String
+						HasNextPage githubv4.Boolean
+					}
+				} `graphql:"reviews(first: $entries, after: $reviewCursor, states: $reviewState)"`
+			} `graphql:"pullRequest(number: $number)"`
+		} `graphql:"repository(owner: $owner, name: $name)"`
+	}
+
+	variables := map[string]interface{}{
+		"owner":        githubv4.String(repo.Owner),
+		"name":         githubv4.String(repo.Name),
+		"number":       githubv4.Int(pull.Num),
+		"entries":      githubv4.Int(10),
+		"reviewState":  []githubv4.PullRequestReviewState{githubv4.PullRequestReviewStateApproved},
+		"reviewCursor": (*githubv4.String)(nil), // Null after argument to get first page.
+	}
+
+	var allReviews []GithubReview
+	for {
```

Comment:
Why does this need to use a for loop? Why can we not retrieve all the pr reviews with a single graphql call to github?

Reply:
This is for pagination. 
The code requests pages of reviews until there are no left.  Currently the page size is 10, so in 99% of cases there should be only a single request as only reviews with the state `Approved` are considered.

Reply:
Ah ok so this is in case there are more than 10 reviews. I got it. Thank you.

Do you mind adding a comment to the code for future readers?